warehouse/packaging/models.py

```diff
@@ -183,6 +188,16 @@ class Project(SitemapMixin, HasEvents, HasObservations, db.Model):
     total_size: Mapped[int | None] = mapped_column(
         BigInteger, server_default=sql.text("0")
     )
+    lifecycle_status: Mapped[LifecycleStatus | None] = mapped_column(
+        comment="Lifecycle status can change project visibility and access"
+    )
+    lifecycle_status_changed: Mapped[datetime_now | None] = mapped_column(
+        onupdate=func.now(),
+        comment="When the lifecycle status was last changed",
+    )
+    lifecycle_status_note: Mapped[str | None] = mapped_column(
+        comment="Note about the lifecycle status"
+    )
```

Comment:
Question: You know way more about schemas and effective layouts than I do, so possibly a dumb question: could this be a `Mapped[Lifecycle]` instead, to avoid the `lifecycle_{attribute}` pattern? Or would that necessitate a `relationship` and therefore be slower 🙂

Reply:
Makes sense, thanks for explaining!